activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/VariablesMappingProvider.java

```diff
@@ -69,13 +69,13 @@ public VariablesMappingProvider(ProcessExtensionService processExtensionService,
 
         Map<String, Object> constants = calculateConstants(execution, extensions);
 
-        if (extensions.getExtensions().hasEmptyInputsMapping(execution.getCurrentActivityId())) {
+        if (extensions.getExtensionsByProcessDefinitionId(execution.getProcessDefinitionId()).hasEmptyInputsMapping(execution.getCurrentActivityId())) {
```

Comment:
I think it's better to return directly an `Extension` instead of `ProcessExtensionModel` as we are already providing the process `definitionId`. Let's modify the signature of `ProcessExtensionService.getExtensionsForId`